hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTaskDbEventHandler.java

```diff
@@ -132,6 +132,10 @@ protected void handlePutDirEvent(OmDirectoryInfo directoryInfo,
       curNSSummary = new NSSummary();
     }
     curNSSummary.setDirName(dirName);
+    // Set the parent directory ID
+    if (parentObjectId != -1) {
+      curNSSummary.setParentId(parentObjectId);
+    }
```

Comment:
what is the scenario for -1 for NSSummary? check for the else part impact

Reply:
I have removed the -1 check from there, I do not see any scenario where the parentID of any directory would be -1 which does not happen anytime, even if that parent is a bucket it will still have a parentID I have double checked it by running the JVM debugger on the cluster code.